nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/renditions/RenditionsObject.java

```diff
@@ -0,0 +1,157 @@
+        query = query != null ? query : PICTURES_DEFAULT_QUERY;
+        String finalQuery = !query.isEmpty() ? query : PICTURES_DEFAULT_QUERY;
+
+        Supplier<BulkStatus> supplier = () -> {
+            if (needToRun(bulkService, commandId)) {
+                String newCommandId = bulkService.submit(new BulkCommand.Builder(RecomputeViewsAction.ACTION_NAME,
+                        finalQuery).user(ctx.getPrincipal().getName())
+                                   .param(RecomputeViewsAction.PARAM_XPATH, "file:content")
+                                   .build());
+                putCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID, newCommandId);
+                return bulkService.getStatus(newCommandId);
+            }
+            throw new WebApplicationException(Response.Status.CONFLICT);
+        };
+        return LockHelper.doAtomically(RECOMPUTE_VIEWS_ACTION_COMMAND_ID, supplier);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doGetPictures() {
+        BulkService service = Framework.getService(BulkService.class);
+        return service.getStatus(getCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID));
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("thumbnails/recompute")
+    public BulkStatus doPostThumbnails(@FormParam("query") String query) throws OperationException {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String commandId = getCommandId(RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID);
+        query = query != null ? query : THUMBNAILS_DEFAULT_QUERY;
+        final String finalQuery = !query.isEmpty() ? query : THUMBNAILS_DEFAULT_QUERY;
+        Supplier<BulkStatus> supplier = () -> {
+            if (needToRun(bulkService, commandId)) {
+                String newCommandId = bulkService.submit(
+                        new BulkCommand.Builder(RecomputeThumbnailsAction.ACTION_NAME, finalQuery).user(
+                                ctx.getPrincipal().getName()).build());
+                putCommandId(RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID, newCommandId);
+                return bulkService.getStatus(newCommandId);
+            }
+            throw new WebApplicationException(Response.Status.CONFLICT);
+        };
+        return LockHelper.doAtomically(RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID, supplier);
+    }
+
+    protected String getCommandId(String commandName) {
+        return getKVStore().getString(commandName);
+    }
+
+    protected void putCommandId(String commandName, String commandId) {
+        getKVStore().put(commandName, commandId);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("thumbnails/recompute")
+    public BulkStatus doGetThumbnails() {
+        BulkService service = Framework.getService(BulkService.class);
+        return service.getStatus(getCommandId(RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID));
```

Comment:
Here, if `getCommandId` returns `null`, you'll have an NPE (so this is probably not covered by the tests).
See https://github.com/nuxeo/nuxeo/blob/master/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java#L151
and
https://github.com/nuxeo/nuxeo/blob/master/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/message/BulkStatus.java#L139
```
public static BulkStatus unknownOf(@NotNull String commandId) {
  ...
}
```
I think that you should first get the command id and handle its nullity, as you are doing it for the POST in `needToRun`.

If it is null, I would throw a
```
WebApplicationException(Response.Status.NOT_FOUND)
```
thus returning a 404 status code.
@kevinleturc @bdelbosc WDYT?

Reply:
Agree 👍.

For my information, do we prefer `WebApplicationException` or a `NuxeoException` with appropriate http code?

Reply:
For now the `WebApplicationException` I throw shows the stack trace to the client.

Reply:
I think that we don't care as the `WebEngineExceptionMapper` handles both the same way:
https://github.com/nuxeo/nuxeo/blob/master/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/WebEngineExceptionMapper.java#L95